<launch>	
<!--
    <include file="$(find openni2_launch)/launch/openni2.launch" />
-->
	<include file="$(find moveit_generated)/launch/wm_move_group.launch" />

  	<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen" />
	
	<node name="wm_arm_driver_node" pkg="wm_arm_driver" type="wm_arm_driver_node" output="screen" >

		<param name="urdf_param" value="/sara" />
		<param name="base_move_group_name" value="base" />
		<param name="manipulator_move_group_name" value="manipulator" />

		<!-- max_velocity in degrees/second -->
		<param name="max_velocity_joints123" value="5" />
		<param name="max_velocity_joints456" value="5" />
		<param name="joint_movement_synchronization" value="true" />

		<!-- home_position and joints_offset must be in degrees -->
		<param name="home_position" value="[0.0, 0.0, 0.0, 0.0, 0.0]" />
		<param name="joints_offset" value='[30.55, 178.73, 202.89, 351.07, 212.86, 0.0]' />		

		<param name="max_torque_joint1" value="20" />
		<param name="max_torque_joint2" value="20" />
		<param name="max_torque_joint3" value="20" />
		<param name="max_torque_joint4" value="20" />
		<param name="max_torque_joint5" value="20" />
		<param name="max_torque_joint6" value="20" />

		<!-- callbacks rate, in Hertz -->
		<param name="torque_callback_rate" value="20" />
		<param name="position_callback_rate" value="20" />
		<param name="velocity_callback_rate" value="20" />

		<!-- planner parameters -->
		<param name="planning_time" value="20" />
		<param name="planning_attempts" value="1" />
		<param name="goal_position_tolerance" value="0.01" />
		<param name="goal_orientation_tolerance" value="0.1" />

		<!-- action server parameters -->
		<param name="action_server_name" value="execute_plan" />
		<param name="action_feedback_rate" value="5" />
		<param name="wait_duration" value="0.5" />
		<param name="maximum_skip" value="2" />

	</node>
</launch>
